src/pudl/extract/phmsagas.py

```diff
@@ -0,0 +1,84 @@
+"""Retrieves data from PHMSA natural gas spreadsheets for analysis.
+
+This modules pulls data from PHMSA's published Excel spreadsheets.
+"""
+
+
+from dagster import AssetOut, Output, multi_asset
+
+import pudl.logging_helpers
+from pudl.extract import excel
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+class Extractor(excel.GenericExtractor):
+    """Extractor for the excel dataset PHMSA."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the module.
+
+        Args:
+            ds (:class:datastore.Datastore): Initialized datastore.
+        """
+        self.METADATA = excel.Metadata("phmsagas")
+        self.cols_added = []
+        super().__init__(*args, **kwargs)
+
+    def process_final_page(self, df, page):
+        """Drop columns that get mapped to other assets.
+
+        Older years of PHMSA data may have fewer pages and tables of data than newer
+        data. This means one page may get split into multiple tables by column. To
+        prevent each table from containing all columns from these older years, filter by
+        the list of columns specified for the page, with a warning.
```

Comment:
This last sentence is still a bit confusing, but I think I'm starting to understand better. Is the idea that the content from one page (tab) gets split into multiple tables because the content in later years is spread across several pages (tabs). And we don't want to retain duplicate columns across tables, one of which will contain data and one of which will be empty?

Reply:
I tried to clarify again.